### PR TITLE
Support typed value in the body of POST request

### PR DESCRIPTION
**Pull request by JungHsuan**
### Purpose:
1. Support typed value instead of converting the value to String.
2. A small refactor has been made to align the code between `WTracker` and `WIdentify`

### Expected Behavior:
Both `WTracker` and `WIdentify` are still functioning as they did before.

### How To Verify it:
Make a push call.
```swift
WTracker.shared.visitor.add(property: "name", value: "new-user-name")
WTracker.shared.visitor.add(property: "email", value: "new-email@mail.com")
WTracker.shared.push()
```

Start a track event and check if it appears on the dashboard.
```swift
let event = WEvent.event(name: "event-name")
event.add(property: "view", value: "view-value")
WTracker.shared.trackEvent(event)
```

### CODE REVIEW

File: WoopraSDK/WIdentify.swift [hunk 1]
```diff
@@ -35,37 +36,39 @@ internal class WIdentify {
...
-            options: []
-        )
-        urlRequest.setValue(
-            "application/json",
-            forHTTPHeaderField: "Content-Type"
-        )
-        
-        print("Request Body: \(requestBody)")
+        // Convert requestBody to JSON data
+        do{
```
Comment (cedricshih) [inline WoopraSDK/WIdentify.swift]: nits: `do {`

File: WoopraSDK/WIdentify.swift [hunk 2]
```diff
@@ -35,37 +36,39 @@ internal class WIdentify {
...
+            let jsonData = try JSONSerialization.data(withJSONObject: requestBody, options:[])
+            request.httpBody = jsonData
+            
+            #if DEBUG
+            if let requestBodyString = String(data: jsonData, encoding: .utf8) {
+                print("Request Body: \(requestBodyString)")
+            }
+            #endif
+        } catch{
```
Comment (cedricshih) [inline WoopraSDK/WIdentify.swift]: nits: `} catch {`

File: WoopraSDK/WIdentify.swift [hunk 3]
```diff
@@ -35,37 +36,39 @@ internal class WIdentify {
...
+        }
         
-        let task = URLSession.shared.dataTask(with: urlRequest) {
-            (
-                data,
-                response,
-                error
-            ) in
+        let task = URLSession.shared.dataTask(with: request) {
+            ( data, response, error) in
```
Comment (cedricshih) [inline WoopraSDK/WIdentify.swift]: nits: `(data, ...`

File: WoopraSDK/WTracker.swift [hunk 4]
```diff
@@ -74,65 +74,62 @@ public class WTracker: WPropertiesContainer {
...
-        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
-        
-        // Convert queryItems to a dictionary
-        let parameters = queryItems.reduce(into: [String: String]()) { result, item in
-            result[item.name] = item.value
-        }
-        if let jsonString = try? JSONSerialization.data(withJSONObject: parameters, options: []) {
-            request.httpBody = jsonString
+        // Convert requestBody to JSON data
+        do{
```
Comment (cedricshih) [inline WoopraSDK/WTracker.swift]: nits: `do {`

File: WoopraSDK/WTracker.swift [hunk 5]
```diff
@@ -74,65 +74,62 @@ public class WTracker: WPropertiesContainer {
...
-                print("Request Body: \(requestBody)")
-            } else {
-                print("Failed to convert request body to string")
+            if let requestBodyString = String(data: jsonData, encoding: .utf8) {
+                print("Request Body: \(requestBodyString)")
             }
             #endif
-        } else {
-            print("Failed to serialize parameters to JSON")
+        } catch{
```
Comment (cedricshih) [inline WoopraSDK/WTracker.swift]: nits: `} catch {`

**Review by wubai-wu-appier — APPROVED** [review]
Approved with behavior review.

**Review by michaelrevlis — APPROVED** [review]
Approved with behavior review

File: WoopraSDK/WIdentify.swift [hunk 6]
```diff
@@ -35,37 +36,44 @@ internal class WIdentify {
...
         
-        urlRequest.httpBody = try? JSONSerialization.data(
-            withJSONObject: requestBody,
-            options: []
-        )
-        urlRequest.setValue(
-            "application/json",
-            forHTTPHeaderField: "Content-Type"
-        )
+        if !JSONSerialization.isValidJSONObject(requestBody) {
```
Comment (michaelrevlis) [inline WoopraSDK/WIdentify.swift]: NIT: prefer `JSONSerialization.isValidJSONObject(requestBody) == false`

File: WoopraSDK/WTracker.swift [hunk 7]
```diff
@@ -67,71 +67,73 @@ public class WTracker: WPropertiesContainer {
...
         }
         
-        var request = URLRequest(url: url)
-        request.httpMethod = "POST"
-        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
-        
-        // Convert queryItems to a dictionary
-        let parameters = queryItems.reduce(into: [String: String]()) { result, item in
-            result[item.name] = item.value
+        if !JSONSerialization.isValidJSONObject(requestBody) {
```
Comment (michaelrevlis) [inline WoopraSDK/WTracker.swift]: NIT: prefer `JSONSerialization.isValidJSONObject(requestBody) == false`

**Pull request merged**